app/models/forumthread.rb

```diff
@@ -86,7 +86,7 @@ def self.filter (user, title, content, reply, label, author, query, forum)
     .joins("LEFT JOIN roles as forumgroup_role_read ON forumgroups.role_read_id = forumgroup_role_read.id")
     .joins("LEFT JOIN roles as forumgroup_role_write ON forumgroups.role_write_id = forumgroup_role_write.id")
 
-    threads = threads.where("forumthreads.user_author_id = ? OR (#{can_read}) OR (#{sticky_can_write})", user_id, role_value, role_value, role_value, role_value)
+    threads = threads.where("forumthreads.user_author_id = ? OR (#{can_read}) OR (#{sticky_can_write}) OR (?)", user_id, role_value, role_value, role_value, role_value, Forum.find(forum).can_read?(user))
```

Comment:
Since this results in:

- `OR (true)` (which will always match) or
- `OR (false)` (which will have no effect on the current query)

you could just wrap the query in a condition and not filter them any further.
Also, given that `(#{can_read})` just mimics `forum.can_read?` in the SQL query, you don't need to include it anymore:
```ruby
unless forum.can_read?(user)
  threads = threads.where("forumthreads.user_author_id = ? OR (#{sticky_can_write})", user_id, role_value, role_value)
end
```